### PR TITLE
[EBPF] Report complexity changes in PRs

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR uses the complexity data generated in CI (PR https://github.com/DataDog/datadog-agent/pull/27274) to report the complexity changes introduced by a PR, using Github PR comments. This data is presented in several ways:

* A general summary message that will tell the developer whether the complexity has improved, is stable or if it has worsened. This helps to signal when the comment needs detailed attention and when it can just be skimmed.
* A list of the most significant figures: highest complexity increases, and programs that went over the threshold of 85% of the instruction limit.
* For each object file with complexity changes, a collapsible table of each program, the change in complexity and the distros with highest/lowest complexity.

### Motivation

Related to https://datadoghq.atlassian.net/browse/EBPF-492, surfacing this data in CI will help developers notice when the complexity of their programs is increasing, so they can remediate it before it becomes a blocker for development.

### CODE REVIEW

File: .gitlab/kernel_matrix_testing/common.yml [hunk 1]
```diff
@@ -268,3 +268,54 @@
...
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner$TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX:$TEST_INFRA_DEFINITIONS_BUILDIMAGES
+  tags: ["arch:amd64"]
+  rules:
+    - !reference [.on_system_probe_or_e2e_changes_or_manual]
+    - !reference [.on_security_agent_changes_or_manual]
+    - !reference [.except_mergequeue]
+    - !reference [.except_main_or_release_branch]
+    - !reference [.except_no_tests_no_deploy]
+  needs:
+    # We need to specify the jobs that generate complexity explicitly, else we hit the limit of "needs"
```
Comment (brycekahle) [inline .gitlab/kernel_matrix_testing/common.yml:283]: Do you still hit the limit if you just specify `kmt_run_sysprobe_tests_x64` and not the individual parts of the matrix?

Comment (gjulianm) [reply .gitlab/kernel_matrix_testing/common.yml:283]: If we want just the `kmt_run_sysprobe_tests_x64` we wouldn't, but then we'd lose the `arm64` data.

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by brycekahle — APPROVED** [review]

**Pull request merged**